### ISSUE TITLE
Better synchronization GUI/GDB.

### ISSUE THREAD

**Issue by perlindgren** [Marus/cortex-debug#114]
When using the  `gdb` console to give commands, at times the `gdb` state is updated, but not the GUI view. E.g. setting breakpoints in the "postLaunchCommands", works well (`gdb` breaks accordingly), however the GUI does not recognize the breakpoint.

I'm not sure if there is an eventing mechanism allowing to register a listener (for `gdb` state changes) or if the plugin would need to probe for state changes.

One can live with the breakpoint issue (as gdb behaves as intended), but it would be nice if fixable.

A more severe issue is if you continue from the (gdb c), something goes wrong. The target cannot be stopped, neither from the GUI nor by giving the gdb command interrupt, (the gdb command interrupt, normally breaks/stops the target). 

The gdb console (and scripting) is super useful when doing (advanced) debugging, so in general, better support/integration would be very good. (It is not only `cortex-debug` that lacks good gdb console integration, the same can be said for Eclipse/Atollic/.. as well and CLion (been a while since I looked at them so they might have improved...not sure).

/Per

**Comment by Marus** [comment]
Unfortunately a lot of this may be down to limitations in the GDB machine interface. While a number of things are reported as events in the machine interface stream - generally user interactions are not. Will look into it more for sure though.

**Comment by perlindgren** [comment]
Hi Marus

Have you had any further success regarding the GDB machine interface interaction?

My students are up and running using your debug plugin, looks good so far. Also we have announced a Workshop on Real-Time for The Masses framework, at the upcoming Oxidize (Rust) conference (end April in Berlin. We will demonstrate the cortex-debug integration among other things...

So we hope for a new release with better integration would be possible till then (and the disassembly view that I mentioned previously needs some love..., its quite sketchy as of now). I saw there was a PR for a better mem view with scrolling, maybe the same approach could be used for disassembly, showing the current PC, and able to scroll up and down... (Similar to Eclipse...)

/Per
